dvc/command/add.py

```diff
@@ -1,22 +1,30 @@
 from __future__ import unicode_literals
 
+import os
 import dvc.logger as logger
 from dvc.exceptions import DvcException
 from dvc.command.base import CmdBase
 
 
 class CmdAdd(CmdBase):
     def run(self):
-        for target in self.args.targets:
-            try:
+        try:
+            if (len(self.args.targets) > 1 and self.args.file) or (
+                os.path.isdir(self.args.targets[0]) and self.args.file
```

Comment:
And what about `dvc add directory -f fname`? :slightly_smiling_face:  It should be allowed, right? If so, we should check for `self.args.recursive` instead of ` os.path.isdir(self.args.targets[0])`. Need to add a test for it as well.

One more thing. What if dvc is called through the API with `dvc.add('directory', recursive=True, fname='somehting.dvc')`? We should not allow it as well and raise an exeption. Could you please add a test for it as well?

Reply:
![image](https://user-images.githubusercontent.com/7363250/53778790-97553780-3ec3-11e9-8169-072dcef83959.png)
Nice catch, @efiop , I'll submit the changes in a moment